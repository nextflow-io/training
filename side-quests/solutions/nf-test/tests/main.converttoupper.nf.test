nextflow_process {

    name "Test Process convertToUpper"
    script "main.nf"
    process "convertToUpper"

    test("Should run without failures and produce correct output") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = "${projectDir}/greetings.csv"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
