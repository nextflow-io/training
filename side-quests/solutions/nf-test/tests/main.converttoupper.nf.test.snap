{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "UPPER-greetings.csv:md5,f36624b0e040fa880e61fb1304b4b6b9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-19T11:57:20.215132756"
    }
}
